cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -671,10 +622,16 @@ struct ConvertFunctor {
                                                       parse_options_view const& opts,
                                                       bool as_hex)
   {
-    T const value                    = decode_value<T>(begin, end, opts);
-    static_cast<T*>(out_buffer)[row] = value;
+    auto const value = [&opts, begin, end]() -> std::optional<T> {
+      // Check for user-specified true/false values
+      auto const field_len = static_cast<size_t>(end - begin);
+      if (serialized_trie_contains(opts.trie_true, {begin, field_len})) { return true; }
+      if (serialized_trie_contains(opts.trie_false, {begin, field_len})) { return false; }
+      return cudf::io::parse_numeric<T>(begin, end, opts);
+    }();
+    static_cast<T*>(out_buffer)[row] = value.value_or(std::numeric_limits<T>::quiet_NaN());
```

Comment:
I'm confused. The PR description says that we switched from returning NaN to returning null, but here it looks like we're still inserting a `quiet_NaN`. What am I missing?

Reply:
This `quiet_NaN` write need not happen if this row is null.  When the row is null, the value held in respective memory location does not matter. But it's writing the value anyway. I felt the reason to leave `NaN` value in the memory location because float has a `NaN` representation, but other types doesn't. I will remove this.

Reply:
moved this code to PR https://github.com/rapidsai/cudf/pull/12022 and updated as per this comment

Reply:
@vyasr CSV_TEST
`CsvReaderTest.InvalidFloatingPoint` expects the value to be nan regardless of bitmask.
https://github.com/karthikeyann/cudf/blob/fc5b67517765f5ebc94abead4b4b6c8cbd43d4da/cpp/tests/io/csv_test.cpp#L1191-L1194

Should this test be updated to ignore the values, and check only bitmask?

Reply:
Yes I think so. Checking the data in a null row is UB.